### PR TITLE
Update Azure Function integration article

### PR DESCRIPTION
**Pull request by hammerhead**
## Summary of the changes / Why this is an improvement
* Added a few more hints on where to find certain settings/what exactly to change in the config files
* Provide a concrete test payload
* There were a few issues with the actual function definition:
  * Some undefined variables were used
  * Timestamps were not handled correctly
  * The structure of the `context` object likely has changed over time

## Checklist

 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed

### CODE REVIEW

File: docs/integrations/azure-functions.rst [hunk 1]
```diff
@@ -238,8 +245,9 @@ you will ingest manned and unmanned satellite data into, respectively.
 These settings **are not** deployed when you deploy the Azure Function. Instead,
 before you deploy, you should set up an Azure Function Application within the
 Azure Portal. You can do this `directly from VSCode`_, or using the
-`Azure CLI`_. These settings would then need to be set up within the Azure
-Function's **Application Settings**.
+`Azure CLI`_. All settings shown above within the `Values` object need to be set
```
Comment (amotl) [inline docs/integrations/azure-functions.rst:248]:
```suggestion
`Azure CLI`_. All settings shown above within the ``Values`` object need to be set
```

File: docs/integrations/azure-functions.rst [hunk 2]
```diff
@@ -271,37 +278,31 @@ can go through it step by step:
 	    query_timeout: 30000,
 	});
 
-
-	// This is the our Azure Function that will be called when new event hub messages are processed.
+	// This is our Azure Function that will be called when new event hub messages are processed.
```
Comment (amotl) [inline docs/integrations/azure-functions.rst:281]:
```suggestion
	// This is the Azure Function that will be called when new event hub messages are processed.
```

File: docs/integrations/azure-functions.rst [hunk 3]
```diff
@@ -351,13 +351,31 @@ can go through it step by step:
...
+the documentation's `debugging`_ section for details.
+The following JSON document can be used as a test message:
+
+.. code-block:: javascript
+
+  {"input": "{\"id\": \"Zero Gravitas\", \"type\": \"unmanned\", \"location\": {\"longitude\": -164.5984,\"latitude\": -24.9734},\"timestamp\": 1588240576000}"}
+
+To test the deployed Azure Function against an actual IoT hub, you can install
+VSCode's `Azure IoT Hub extension`_. Its documentation describes how to create
+a new device and send a device-to-cloud (D2C) message for test purposes.
```
Comment (amotl) [inline docs/integrations/azure-functions.rst:367]:
```suggestion
a new device and send a device-to-cloud (D2C) message for testing purposes.
```

**Review by amotl — CHANGES_REQUESTED** [review]
Hi Niklas,

thanks a stack for giving this piece some love - both for improving the prose and for fixing code issues. I added some comments as minor suggestions.

With kind regards,
Andreas.

**Comment by hammerhead** [comment]
Thanks for the feedback @amotl, I've pushed an updated version with your changes incorporated.

**Review by amotl — APPROVED** [review]
👍

File: docs/integrations/azure-functions.rst [hunk 4]
```diff
@@ -223,13 +226,17 @@ project. It should have the following content:
 	    }
 	  }
 
-The ``AzureWebJobsStorage`` and ``EventHubConnectionString`` connection strings
-should be replaced with the relevant ones that you can find in the Azure
-Portal, ensuring that the event hub connection string includes the
+The ``AzureWebJobsStorage`` value can be obtained by navigating to your storage
+account, selecting "Access keys" and copying one of the shown connection
```
Comment (matthijskrul) [inline docs/integrations/azure-functions.rst:230]:
```suggestion
account, selecting "Access keys", and copying one of the shown connection
```

File: docs/integrations/azure-functions.rst [hunk 5]
```diff
@@ -223,13 +226,17 @@ project. It should have the following content:
...
+strings.
+The ``EventHubConnectionString`` can be copied from the
```
Comment (matthijskrul) [inline docs/integrations/azure-functions.rst:232]:
```suggestion

The ``EventHubConnectionString`` can be copied from the
```

File: docs/integrations/azure-functions.rst [hunk 6]
```diff
@@ -238,8 +245,9 @@ you will ingest manned and unmanned satellite data into, respectively.
 These settings **are not** deployed when you deploy the Azure Function. Instead,
 before you deploy, you should set up an Azure Function Application within the
 Azure Portal. You can do this `directly from VSCode`_, or using the
-`Azure CLI`_. These settings would then need to be set up within the Azure
-Function's **Application Settings**.
+`Azure CLI`_. All settings shown above within the ``Values`` object need to be set
+up within the Azure Function App's **Application Settings** which can be found
```
Comment (matthijskrul) [inline docs/integrations/azure-functions.rst:249]:
```suggestion
up within the Azure Function App's **Application Settings**, which can be found
```

File: docs/integrations/azure-functions.rst [hunk 7]
```diff
@@ -260,7 +268,6 @@ can go through it step by step:
 	const SINK_COLUMN_MANNED = process.env['SinkColumnManned'];
 	const SINK_COLUMN_DEBUG = "debug";
 
-
 	// A pool of connections to CrateDB that our Azure Function can utilise.
```
Comment (matthijskrul) [inline docs/integrations/azure-functions.rst:271]:
```suggestion
	// A pool of connections to CrateDB that our Azure Function can utilize.
```

**Review by matthijskrul — APPROVED** [review]

**Comment by matthijskrul** [comment]
Dang you were just a few seconds too fast! 😁

**Comment by hammerhead** [comment]
Too bad, I'll create a mini follow-up pull request for those missed changes.

**Pull request merged**